### DISCUSSION TITLE
typecheck fails because it can't locate cypress and @testing-library/cypress type definitions

### DISCUSSION THREAD

**Discussion by evanstern** [remix-run/remix#6528] [Q&A] [answered]
This one has me stumped. 

I'm not seeing anyone reporting this issue, so I will assume I'm an idiot 😊. Feel free to tell me I'm doing something very stupid.

Both in GitHub when running the deploy action and locally when running `npm run typecheck` the process errors with the following output:

```
error TS2688: Cannot find type definition file for '@testing-library/cypress'.
  The file is in the program because:
    Entry point of type library '@testing-library/cypress' specified in compilerOptions

  cypress/tsconfig.json:16:34
    16     "types": ["node", "cypress", "@testing-library/cypress"],
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~
    File is entry point of type library specified here.

error TS2688: Cannot find type definition file for 'cypress'.
  The file is in the program because:
    Entry point of type library 'cypress' specified in compilerOptions

  cypress/tsconfig.json:16:23
    16     "types": ["node", "cypress", "@testing-library/cypress"],
                             ~~~~~~~~~
    File is entry point of type library specified here.


Found 2 errors.
```

The `cypress/tsconfig.json` file seems correct to me. The `"include"` section does include the cypress and testing-library/cypress locations from the parent's node_modules directory:

```
{
  "exclude": [
    "../node_modules/@types/jest",
    "../node_modules/@testing-library/jest-dom"
  ],
  "include": [
    "e2e/**/*",
    "support/**/*",
    "../node_modules/cypress",
    "../node_modules/@testing-library/cypress"
  ],
  "compilerOptions": {
    "baseUrl": ".",
    "noEmit": true,
    "types": ["node", "cypress", "@testing-library/cypress"],
    "esModuleInterop": true,
    "jsx": "react-jsx",
    "moduleResolution": "node",
    "target": "es2019",
    "strict": true,
    "skipLibCheck": true,
    "resolveJsonModule": true,
    "typeRoots": ["../types", "../node_modules/@types"],

    "paths": {
      "~/*": ["../app/*"]
    }
  }
}
```

I have also triple-checked that the packages are installed and where they're expected to be.

Has anyone seen this before?

**Comment by Hiturunk** [comment]
Thanks, was also struggling with this today. :+1:

**Comment by evanstern** [comment]
I was able to identify the issue and resolve it.

First of all, this is happening for me in the Indie Stack.

It looks like the culprit is the typescript dependency at version ^5

Downgrading to version ^4.9.5 solved the issue.

That may not be a long term solution, but if you just want to get typecheck to work, it does seem to fix things.

**Comment by MichaelDeBoey** [comment, marked as answer]
Duplicate of https://github.com/remix-run/indie-stack/issues/239
